packages/model-viewer/karma.conf.js

```diff
@@ -0,0 +1,206 @@
+    }
+
+    // This terrible hack brought to you by a combination of two things:
+    //  1. BrowserStack drops the server port when redirecting from localhost
+    //     to bs-local.com on iOS
+    //  2. karma-browserstack-launcher drops the test-specific browser ID if
+    //     you configure the browser with a custom URL
+    // A support request to BrowserStack has been filed.
+    // A related bug has been filed againts karma-browserstack-launcher
+    // @see https://github.com/karma-runner/karma-browserstack-launcher/issues/172
+    const assign = Object.assign;
+    const newAssign = function(...args) {
+      // If we know this to be one very specific Object.assign call, then grab
+      // the test-specific browser ID and append it to our URL:
+      // @see https://github.com/karma-runner/karma-browserstack-launcher/blob/76dbfd0db6db46f4f85012cfe3c1f4c3accd2e44/index.js#L143
+      if (args[2] != null && args[2].url === 'http://bs-local.com:9876') {
+        console.warn('PATCHING URL TO ADD ID');
+        const config = args[0];
+        const browser = args[2];
+        const query = config.url.split('?')[1];
+        browser.url = `${browser.url}?${query}`;
+      }
+      return assign.apply(this, args);
+    };
+    // Something in Karma deps actually asserts the sub-keys of Object.assign,
+    // so make sure to copy those over too:
+    assign.call(Object, newAssign, assign);
+    Object.assign = newAssign;
+
+    const browserStackLaunchers = {
+      'Edge (latest)': {
+        base: 'BrowserStack',
+        os: 'Windows',
+        os_version: '10',
+        browser: 'Edge',
+        browser_version: 'latest',
+      },
+      'Edge 79.0': {
+        base: 'BrowserStack',
+        os: 'Windows',
+        os_version: '10',
+        browser: 'Edge',
+        browser_version: '80.0',
+      },
+      'Firefox (latest)': {
+        base: 'BrowserStack',
+        os: 'Windows',
+        os_version: '10',
+        browser: 'Firefox',
+        browser_version: 'latest',
+      },
+      'Firefox 72.0': {
+        base: 'BrowserStack',
+        os: 'Windows',
+        os_version: '10',
+        browser: 'Firefox',
+        browser_version: '72.0',
+      },
+      'Safari (latest)': {
+        base: 'BrowserStack',
+        os: 'OS X',
+        os_version: 'Catalina',
+        browser: 'safari',
+        browser_version: 'latest',
+      },
+      'Safari 12.1': {
+        base: 'BrowserStack',
+        os: 'OS X',
+        os_version: 'Mojave',
+        browser: 'safari',
+        browser_version: '12.1',
+      },
+      'iOS Safari (iOS 13)': {
+        base: 'BrowserStack',
+        os: 'iOS',
+        os_version: '12',
+        device: 'iPhone 8',
+        browser: 'iPhone',
+        real_mobile: 'true',
+        // BrowserStack seems to drop the port when redirecting to this special
+        // domain so we go there directly instead:
+        url: 'http://bs-local.com:9876'
+      },
+      'iOS Safari (iOS 12)': {
```

Comment:
Should we add an android device someday? Not saying today is that day...

Reply:
Yes, let's make it another day though. Tracked in #691